resources/stsci.edu/schemas/transform-1.2.0.yaml

```diff
@@ -9,22 +9,80 @@ description: >
   These objects are designed to be nested in arbitrary ways to build up
   transformation pipelines out of a number of low-level pieces.
 
-type: object
-properties:
-  name:
-    description: |
-      A user-friendly name for the transform, to give it extra
-      meaning.
-    type: string
+examples:
+  -
+    - A transform with no bounding_box or compound_bounding_box
+    - |
+      !transform/transform-1.2.0
+        name: test
+  -
+    - A transform with 1D bounding_box
+    - |
+      !transform/transform-1.2.0
+        name: test
+        bounding_box: [1.0, 2.0]
+  -
+    - A transform with 2D bounding_box
+    - |
+      !transform/transform-1.2.0
+        name: test
+        bounding_box:
+          - [1.0, 2.0]
+          - [3.0, 4.0]
+  -
+    - A transform with compound_bounding_box
+    - |
+      !transform/transform-1.2.0
+        name: test
+        compound_bounding_box:
+          selector_args:
+            - argument: x
+              ignored: true
+            - argument: y
+              ignored: true
+          cbbox:
+            - key: [0, 1]
+              bbox:
+                - [2.0, 3.0]
+                - [4.0, 5.0]
+            - key: [6, 7]
+              bbox:
+                - [8.0, 9.0]
+                - [10.0, 11.0]
 
-  inverse:
-    description: |
-      Explicitly sets the inverse transform of this transform.
+definitions:
+  base_transform:
+    type: object
+    properties:
+      name:
+        description: |
+          A user-friendly name for the transform, to give it extra
+          meaning.
+        type: string
 
-      If the transform has a direct analytic inverse, this
-      property is usually not necessary, as the ASDF-reading tool
-      can provide it automatically.
+      inverse:
+        description: |
+          Explicitly sets the inverse transform of this transform.
 
-    $ref: "transform-1.2.0"
-additionalProperties: true
+          If the transform has a direct analytic inverse, this
+          property is usually not necessary, as the ASDF-reading tool
+          can provide it automatically.
+
+        $ref: "transform-1.2.0"
+    additionalProperties: true
+
+allOf:
+  - $ref: "#/definitions/base_transform"
+  - oneOf:
+    - properties:
+        bounding_box:
+          $ref: bounding_box-1.0.0
+      required: [bounding_box]
+    - properties:
+        compound_bounding_box:
+          $ref: compound_bounding_box-1.0.0
+      required: [compound_bounding_box]
+    - properties:
+        bounding_box: {not: {}}
```

Comment:
This trick of the last case to ensure that having neither bounding_box or compound_bounding_box is both nonintuitive  and non-obvious that it deserves to be attended to with an explanatory comment about its purpose. "This is needed to permit that having neither property is legal."

Reply:
This trick is no longer needed. Commit 95f4d35 started introducing changes to avoid needing this trick. All bounding box like objects will be listed under the `bounding_box` keyword.